File: 389.找不同/389.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        //整体逻辑是先把两个字符串排序，然后比较每个位置的字符，如果不一样了就说明这个位置的字符就是多出来的那个字符
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());
        if(s.length()==0) return t[0];
        //如果s字符串长度为0了，说明t字符串的第一个字符就是多出来的那个字符
        char word=t.back();//如果s字符串长度不为0了，说明t字符串的最后一个字符就是多出来的那个字符
        for(int i=0;i<s.length();i++){
            if(s[i]!=t[i]){
                word=t[i];//如果s字符串和t字符串在某个位置的字符不一样了，说明这个位置的字符就是多出来的那个字符
                break;//为了word的值不被后面t字符串的字符覆盖了，找到了就break掉循环，不再继续比较了
            }
        }
        return word;
    }
};

File: 389.找不同/389.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        ret=0
        for ch in s:
            ret^=ord(ch)
            #ord函数可以把字符转换成对应的ASCII码值，^是异或运算符，异或运算符的特点是相同的数异或结果为0，
            # 不同的数异或结果为1，所以我们先把s字符串中的每个字符都异或一遍，最后再把t字符串中的每个字符都异或一遍，
            # 最后ret中就只剩下了多出来的那个字符的ASCII码值了
        for ch in t:
            ret^=ord(ch)
        return chr(ret)
        